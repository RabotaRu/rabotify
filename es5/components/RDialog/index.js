'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _RDialog = require('./RDialog.vue');

var _RDialog2 = _interopRequireDefault(_RDialog);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* istanbul ignore next */
_RDialog2.default.install = function install(Vue) {
  Vue.component('r-dialog', _RDialog2.default);
};

exports.default = _RDialog2.default;