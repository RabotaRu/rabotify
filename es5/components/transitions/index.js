'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RExpandTransition = exports.RScaleTransitionGroup = exports.RSlideYReverseTransitionGroup = exports.RSlideYTransitionGroup = exports.RSlideXReverseTransitionGroup = exports.RSlideXTransitionGroup = exports.RFadeTransitionGroup = exports.RSlideYReverseTransition = exports.RSlideYTransition = exports.RSlideXReverseTransition = exports.RSlideXTransition = exports.RScaleTransition = exports.RFadeTransition = exports.RToastTransition = exports.RFabTransition = exports.RMenuTransition = exports.RTabReverseTransition = exports.RTabTransition = exports.RCarouselReverseTransition = exports.RCarouselTransition = exports.RBottomSheetTranstion = exports.RDialogRightTransition = exports.RDialogLeftTransition = exports.RDialogBottomTransition = exports.RDialogTopTransition = exports.RDialogTransition = undefined;

var _helpers = require('../../util/helpers');

var _expandTransition = require('./expand-transition');

var _expandTransition2 = _interopRequireDefault(_expandTransition);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// Component specific transitions
var RDialogTransition = exports.RDialogTransition = (0, _helpers.createSimpleTransition)('dialog-transition');
var RDialogTopTransition = exports.RDialogTopTransition = (0, _helpers.createSimpleTransition)('dialog-top-transition');
var RDialogBottomTransition = exports.RDialogBottomTransition = (0, _helpers.createSimpleTransition)('dialog-bottom-transition');
var RDialogLeftTransition = exports.RDialogLeftTransition = (0, _helpers.createSimpleTransition)('dialog-left-transition');
var RDialogRightTransition = exports.RDialogRightTransition = (0, _helpers.createSimpleTransition)('dialog-right-transition');

var RBottomSheetTranstion = exports.RBottomSheetTranstion = (0, _helpers.createSimpleTransition)('bottom-sheet-transition');
var RCarouselTransition = exports.RCarouselTransition = (0, _helpers.createSimpleTransition)('carousel-transition');
var RCarouselReverseTransition = exports.RCarouselReverseTransition = (0, _helpers.createSimpleTransition)('carousel-reverse-transition');
var RTabTransition = exports.RTabTransition = (0, _helpers.createSimpleTransition)('tab-transition');
var RTabReverseTransition = exports.RTabReverseTransition = (0, _helpers.createSimpleTransition)('tab-reverse-transition');
var RMenuTransition = exports.RMenuTransition = (0, _helpers.createSimpleTransition)('menu-transition');
var RFabTransition = exports.RFabTransition = (0, _helpers.createSimpleTransition)('fab-transition', 'center center', 'out-in');
var RToastTransition = exports.RToastTransition = (0, _helpers.createSimpleTransition)('toast-transition', 'center center', 'out-in');

// Generic transitions
var RFadeTransition = exports.RFadeTransition = (0, _helpers.createSimpleTransition)('fade-transition');
var RScaleTransition = exports.RScaleTransition = (0, _helpers.createSimpleTransition)('scale-transition');
var RSlideXTransition = exports.RSlideXTransition = (0, _helpers.createSimpleTransition)('slide-x-transition');
var RSlideXReverseTransition = exports.RSlideXReverseTransition = (0, _helpers.createSimpleTransition)('slide-x-reverse-transition');
var RSlideYTransition = exports.RSlideYTransition = (0, _helpers.createSimpleTransition)('slide-y-transition');
var RSlideYReverseTransition = exports.RSlideYReverseTransition = (0, _helpers.createSimpleTransition)('slide-y-reverse-transition');

// Generic transitions' groups
var RFadeTransitionGroup = exports.RFadeTransitionGroup = (0, _helpers.createSimpleTransitionGroup)('fade-transition');
var RSlideXTransitionGroup = exports.RSlideXTransitionGroup = (0, _helpers.createSimpleTransitionGroup)('slide-x-transition');
var RSlideXReverseTransitionGroup = exports.RSlideXReverseTransitionGroup = (0, _helpers.createSimpleTransitionGroup)('slide-x-reverse-transition');
var RSlideYTransitionGroup = exports.RSlideYTransitionGroup = (0, _helpers.createSimpleTransitionGroup)('slide-y-transition');
var RSlideYReverseTransitionGroup = exports.RSlideYReverseTransitionGroup = (0, _helpers.createSimpleTransitionGroup)('slide-y-reverse-transition');
var RScaleTransitionGroup = exports.RScaleTransitionGroup = (0, _helpers.createSimpleTransitionGroup)('scale-transition');

// JavaScript transitions
var RExpandTransition = exports.RExpandTransition = (0, _helpers.createJavaScriptTransition)('expand-transition', (0, _expandTransition2.default)());

/* istanbul ignore next */
// eslint-disable-next-line max-statements
function install(Vue) {
  // Component transitions
  Vue.component('r-bottom-sheet-transition', RBottomSheetTranstion);
  Vue.component('r-carousel-transition', RCarouselTransition);
  Vue.component('r-carousel-reverse-transition', RCarouselReverseTransition);
  Vue.component('r-dialog-transition', RDialogTransition);
  Vue.component('r-dialog-top-transition', RDialogTopTransition);
  Vue.component('r-dialog-bottom-transition', RDialogBottomTransition);
  Vue.component('r-dialog-left-transition', RDialogLeftTransition);
  Vue.component('r-dialog-right-transition', RDialogRightTransition);
  Vue.component('r-tab-reverse-transition', RTabReverseTransition);
  Vue.component('r-tab-transition', RTabTransition);
  Vue.component('r-fab-transition', RFabTransition);
  Vue.component('r-menu-transition', RMenuTransition);

  // Generic transitions
  Vue.component('r-fade-transition', RFadeTransition);
  Vue.component('r-scale-transition', RScaleTransition);
  Vue.component('r-slide-x-transition', RSlideXTransition);
  Vue.component('r-slide-x-reverse-transition', RSlideXReverseTransition);
  Vue.component('r-slide-y-transition', RSlideYTransition);
  Vue.component('r-slide-y-reverse-transition', RSlideYReverseTransition);

  // Generic group transitions
  Vue.component('r-fade-transition-group', RFadeTransitionGroup);
  Vue.component('r-slide-x-transition-group', RSlideXTransitionGroup);
  Vue.component('r-slide-x-reverse-transition-group', RSlideXReverseTransitionGroup);
  Vue.component('r-slide-y-transition-group', RSlideYTransitionGroup);
  Vue.component('r-slide-y-reverse-transition-group', RSlideYReverseTransitionGroup);
  Vue.component('r-scale-transition-group', RScaleTransitionGroup);

  // JavaScript transitions
  Vue.component('r-expand-transition', RExpandTransition);
}

exports.default = install;