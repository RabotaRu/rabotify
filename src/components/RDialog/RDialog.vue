<script>
  // Mixins
  import Dependent from '../../mixins/dependent';
  import Detachable from '../../mixins/detachable';
  import Overlayable from '../../mixins/overlayable';
  import Stackable from '../../mixins/stackable';
  import Toggleable from '../../mixins/toggleable';
  import Returnable from '../../mixins/returnable';

  // Directives
  import ClickOutside from '../../directives/click-outside';
  import KeydownToggle from '../../directives/keydown-toggle';

  // Helpers
  import { getZIndex } from '../../util/helpers';

  export default {
    name: 'r-dialog',

    mixins: [
      Dependent,
      Detachable,
      Overlayable,
      Returnable,
      Stackable,
      Toggleable
    ],

    directives: {
      ClickOutside,
      KeydownToggle
    },

    data () {
      return {
        isDependent: false,
        stackClass: 'dialog__content__active',
        stackMinZIndex: 200
      };
    },

    props: {
      disabled: Boolean,
      persistent: Boolean,
      fullscreen: Boolean,
      fullWidth: Boolean,
      fullHeight: Boolean,
      attachLeft: Boolean,
      attachRight: Boolean,
      attachTop: Boolean,
      attachBottom: Boolean,
      maxWidth: {
        type: [ String, Number ],
        default: 'none'
      },
      origin: {
        type: String,
        default: 'center center'
      },
      width: {
        type: [ String, Number ],
        default: 'auto'
      },
      scrollable: Boolean,
      escCloseable: {
        type: Boolean,
        default: true
      },
      transition: {
        type: [ String, Boolean ],
        default: 'dialog-transition'
      }
    },

    computed: {
      classes () {
        return {
          'dialog': true,
          'dialog_active': this.isActive,
          'dialog_persistent': this.persistent,
          'dialog_fullscreen': this.fullscreen,
          'dialog_sidebar': ( this.attachRight || this.attachLeft ) && this.fullHeight,
          'dialog_full-height': this.fullHeight,
          'dialog_scrollable': this.scrollable,
          [this.contentClass]: !!this.contentClass
        };
      },
      contentClasses () {
        return {
          'dialog__content': true,
          [this.stackClass]: this.isActive,
          'dialog__content_left': this.attachLeft,
          'dialog__content_right': this.attachRight,
          'dialog__content_top': this.attachTop,
          'dialog__content_bottom': this.attachBottom
        };
      }
    },

    watch: {
      isActive (val) {
        if (val) {
          this.show();
        } else {
          this.removeOverlay();
          this.unbind();
        }
      }
    },

    mounted () {
      this.isBooted = this.isActive;
      this.isActive && this.show();
    },

    beforeDestroy () {
      if (typeof window !== 'undefined') {
        this.unbind();
      }
    },

    methods: {
      canCloseImplicitly () {
        return this.isActive && !this.persistent &&
          getZIndex(this.$refs.content) >= this.getMaxZIndex();
      },
      closeConditional (e) {
        // close dialog if !persistent, clicked outside and we're the topmost dialog.
        // Since this should only be called in a capture event (bottom up), we shouldn't need to stop propagation
        return this.canCloseImplicitly() &&
          !this.$refs.content.contains(e.target);
      },
      show () {
        !this.fullscreen && !this.hideOverlay && this.genOverlay();
        this.fullscreen && this.hideScroll();
        this.$refs.content.focus();
        this.$listeners.keydown && this.bind();
      },
      bind () {
        window.addEventListener('keydown', this.onKeydown);
      },
      unbind () {
        window.removeEventListener('keydown', this.onKeydown);
      },
      onKeydown (e) {
        this.$emit('keydown', e);
      }
    },

    render (h) {
      const directives = [];
      if (this.escCloseable) {
        directives.push({
          name: 'keydown-toggle',
          value: {
            closeConditional: (e) => {
              return this.escCloseable &&
                e.keyCode === 27 &&
                this.canCloseImplicitly();
            },
            callback: () => {
              this.isActive = false;
            }
          }
        });
      }
      const children = [];
      const data = {
        'class': this.classes,
        ref: 'dialog',
        directives: [{
          name: 'click-outside',
          value: () => (this.isActive = false),
          args: {
            closeConditional: this.closeConditional,
            include: this.getOpenDependentElements
          }
        }, {
          name: 'show',
          value: this.isActive
        }, ...directives ],
        on: {
          click: e => {
            e.stopPropagation();
          }
        }
      };

      if (!this.fullscreen) {
        data.style = {
          maxWidth: this.maxWidth === 'none' ? undefined : (isNaN(this.maxWidth) ? this.maxWidth : `${this.maxWidth}px`),
          width: this.width === 'auto' ? undefined : (isNaN(this.width) ? this.width : `${this.width}px`)
        };
      }

      if (this.$slots.activator) {
        children.push(h('div', {
          'class': 'dialog__activator',
          on: {
            click: e => {
              e.stopPropagation();
              if (!this.disabled) {
                this.isActive = !this.isActive;
              }
            }
          }
        }, [ this.$slots.activator ]));
      }

      const dialog = h(`r-${this.transition}`, {
        props: {
          origin: this.origin
        }
      }, [
        h('div', data,
          this.showLazyContent(this.$slots.default)
        )
      ]);

      children.push(h('div', {
        'class': this.contentClasses,
        domProps: { tabIndex: -1 },
        style: { zIndex: this.activeZIndex },
        ref: 'content'
      }, [ dialog ]));

      return h('div', {
        'class': 'dialog__container',
        style: {
          display: !this.$slots.activator && 'none' || this.fullWidth ? 'block' : 'inline-block'
        }
      }, children);
    }
  };
</script>

<style lang="scss">
  @import "../../styles/components/dialog/_dialog.scss";
  @import "../../styles/components/_overlay.scss";
</style>
