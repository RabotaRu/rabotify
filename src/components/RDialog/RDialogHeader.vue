<script>
  export default {
    name: 'r-dialog-header',

    props: {
      flat: Boolean
    },

    computed: {
      classes () {
        return {
          'dialog-header': true,
          'dialog-header_flat': this.flat,
        };
      }
    },

    render (h) {
      const children = [];
      const data = {
        'class': this.classes,
        on: this.$listeners
      };

      children.push(
        h('div', {
          staticClass: 'dialog-header__title',
          ref: 'title'
        }, this.$slots.default)
      );

      return h( 'nav', data, children );
    }
  };
</script>

<style lang="scss">
  @import "../../styles/components/dialog/_dialog-header.scss";
</style>
