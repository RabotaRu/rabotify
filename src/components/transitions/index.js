import {
  createSimpleTransition,
  createSimpleTransitionGroup,
  createJavaScriptTransition
} from '../../util/helpers';

import ExpandTransitionGenerator from './expand-transition';

// Component specific transitions
export const RDialogTransition = createSimpleTransition('dialog-transition');
export const RDialogTopTransition = createSimpleTransition('dialog-top-transition');
export const RDialogBottomTransition = createSimpleTransition('dialog-bottom-transition');
export const RDialogLeftTransition = createSimpleTransition('dialog-left-transition');
export const RDialogRightTransition = createSimpleTransition('dialog-right-transition');

export const RBottomSheetTranstion = createSimpleTransition('bottom-sheet-transition');
export const RCarouselTransition = createSimpleTransition('carousel-transition');
export const RCarouselReverseTransition = createSimpleTransition('carousel-reverse-transition');
export const RTabTransition = createSimpleTransition('tab-transition');
export const RTabReverseTransition = createSimpleTransition('tab-reverse-transition');
export const RMenuTransition = createSimpleTransition('menu-transition');
export const RFabTransition = createSimpleTransition('fab-transition', 'center center', 'out-in');
export const RToastTransition = createSimpleTransition('toast-transition', 'center center', 'out-in');

// Generic transitions
export const RFadeTransition = createSimpleTransition('fade-transition');
export const RScaleTransition = createSimpleTransition('scale-transition');
export const RSlideXTransition = createSimpleTransition('slide-x-transition');
export const RSlideXReverseTransition = createSimpleTransition('slide-x-reverse-transition');
export const RSlideYTransition = createSimpleTransition('slide-y-transition');
export const RSlideYReverseTransition = createSimpleTransition('slide-y-reverse-transition');

// Generic transitions' groups
export const RFadeTransitionGroup = createSimpleTransitionGroup('fade-transition');
export const RSlideXTransitionGroup = createSimpleTransitionGroup('slide-x-transition');
export const RSlideXReverseTransitionGroup = createSimpleTransitionGroup('slide-x-reverse-transition');
export const RSlideYTransitionGroup = createSimpleTransitionGroup('slide-y-transition');
export const RSlideYReverseTransitionGroup = createSimpleTransitionGroup('slide-y-reverse-transition');
export const RScaleTransitionGroup = createSimpleTransitionGroup('scale-transition');

// JavaScript transitions
export const RExpandTransition = createJavaScriptTransition('expand-transition', ExpandTransitionGenerator());

/* istanbul ignore next */
// eslint-disable-next-line max-statements
function install (Vue) {
  // Component transitions
  Vue.component('r-bottom-sheet-transition', RBottomSheetTranstion);
  Vue.component('r-carousel-transition', RCarouselTransition);
  Vue.component('r-carousel-reverse-transition', RCarouselReverseTransition);
  Vue.component('r-dialog-transition', RDialogTransition);
  Vue.component('r-dialog-top-transition', RDialogTopTransition);
  Vue.component('r-dialog-bottom-transition', RDialogBottomTransition);
  Vue.component('r-dialog-left-transition', RDialogLeftTransition);
  Vue.component('r-dialog-right-transition', RDialogRightTransition);
  Vue.component('r-tab-reverse-transition', RTabReverseTransition);
  Vue.component('r-tab-transition', RTabTransition);
  Vue.component('r-fab-transition', RFabTransition);
  Vue.component('r-menu-transition', RMenuTransition);

  // Generic transitions
  Vue.component('r-fade-transition', RFadeTransition);
  Vue.component('r-scale-transition', RScaleTransition);
  Vue.component('r-slide-x-transition', RSlideXTransition);
  Vue.component('r-slide-x-reverse-transition', RSlideXReverseTransition);
  Vue.component('r-slide-y-transition', RSlideYTransition);
  Vue.component('r-slide-y-reverse-transition', RSlideYReverseTransition);

  // Generic group transitions
  Vue.component('r-fade-transition-group', RFadeTransitionGroup);
  Vue.component('r-slide-x-transition-group', RSlideXTransitionGroup);
  Vue.component('r-slide-x-reverse-transition-group', RSlideXReverseTransitionGroup);
  Vue.component('r-slide-y-transition-group', RSlideYTransitionGroup);
  Vue.component('r-slide-y-reverse-transition-group', RSlideYReverseTransitionGroup);
  Vue.component('r-scale-transition-group', RScaleTransitionGroup);

  // JavaScript transitions
  Vue.component('r-expand-transition', RExpandTransition);
}

export default install;
