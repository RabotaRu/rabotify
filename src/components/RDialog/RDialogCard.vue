<script>
  // Directives
  import Scroll from '../../directives/scroll';

  // Icons
  import MdClose from '@rabota/md-svg-vue/dist/navigation/MdClose.vue';

  // Components
  import RBtn from '../RBtn/RBtn.vue';
  import RCard from '../RCard/RCard.vue';
  import RCardText from '../RCard/RCardText.vue';
  import RCardActions from '../RCard/RCardActions.vue';
  import RDialogHeader from './RDialogHeader.vue';

  let dialogCardNumber = 1;

  export default {
    name: 'r-dialog-card',

    components: {
      MdClose,

      RBtn,
      RCard,
      RCardText,
      RCardActions,
      RDialogHeader,
    },

    directives: {
      Scroll
    },

    props: {
      actionsStyles: Object,
      actionsClass: String
    },

    data () {
      return {
        id: dialogCardNumber++,
        headerFlat: true
      };
    },

    methods: {
      onContentScroll (ev) {
        this.headerFlat = !( ev.target && ev.target.scrollTop > 0 );
      },

      closeDialog () {
        this.$emit('close');
      }
    },

    computed: {
      classes () {
        return {
          'dialog-card': true
        };
      },

      dialogHeaderClasses () {
        return {
          'dialog-card__header': true
        };
      },

      actionsClasses () {
        return {
          'dialog-card__actions': true,
          [this.actionsClass]: !!this.actionsClass
        };
      },

      dialogCardId () {
        return `dialog-card-${this.id}`;
      },

      dialogCardScrollTarget () {
        return `#${this.dialogCardId} .dialog-card__inner`;
      }
    }
  };
</script>

<template>
  <r-card :class="classes" :id="dialogCardId">

    <div class="dialog-card__header-wrapper">
      <template v-if="$slots.header">
        <slot name="header"></slot>
      </template>
      <r-dialog-header v-else-if="$slots.title || $slots.close"
                       :class="dialogHeaderClasses"
                       :flat="headerFlat">
        <slot name="title"></slot>
      </r-dialog-header>
    </div>

    <div class="dialog-card__close">
      <r-btn v-if="!$slots.close"
             ripple icon flat round large
             @click.native.stop="closeDialog()">
        <md-close></md-close>
      </r-btn>
      <slot v-else name="close"></slot>
    </div>

    <r-card-text ref="text"
                 class="dialog-card__inner"
                 v-scroll="{ callback: onContentScroll, target: dialogCardScrollTarget }">
      <slot name="default"></slot>
    </r-card-text>

    <r-card-actions ref="actions"
                    v-if="$slots.actions"
                    :class="actionsClasses"
                    :style="actionsStyles">
      <slot name="actions"></slot>
    </r-card-actions>
  </r-card>
</template>

<style lang="scss">
  @import "../../styles/components/dialog/_dialog-card.scss";
</style>
