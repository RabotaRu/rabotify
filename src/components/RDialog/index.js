import RDialog from './RDialog.vue';
import RDialogCard from './RDialogCard.vue';
import RDialogHeader from './RDialogHeader.vue';

export {
  RDialog,
  RDialogCard,
  RDialogHeader
};

/* istanbul ignore next */
RDialog.install = function install (Vue) {
  Vue.component('r-dialog', RDialog);
  Vue.component('r-dialog-card', RDialogCard);
  Vue.component('r-dialog-header', RDialogHeader);
};

export default RDialog;
