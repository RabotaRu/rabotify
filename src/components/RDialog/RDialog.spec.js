import Vue from 'vue';
import transitions from '@components/transitions';
import RDialog from '@components/RDialog';
import { test } from '@util/testing';

Vue.use(transitions);

test('RDialog.vue', ({ mount, compileToFunctions }) => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(RDialog);

    expect(wrapper.html()).toMatchSnapshot();
    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });

  it('should render a disabled component and match snapshot', () => {
    const wrapper = mount(RDialog, {
      propsData: {
        disabled: true
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });

  it('should render a persistent component and match snapshot', () => {
    const wrapper = mount(RDialog, {
      propsData: {
        persistent: true
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });

  it('should render a fullscreen component and match snapshot', () => {
    const wrapper = mount(RDialog, {
      propsData: {
        fullscreen: true
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });

  it('should render a lazy component and match snapshot', () => {
    const wrapper = mount(RDialog, {
      propsData: {
        lazy: true
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });

  it('should render a scrollable component and match snapshot', () => {
    const wrapper = mount(RDialog, {
      propsData: {
        scrollable: true
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });

  it('should render component with custom origin and match snapshot', () => {
    const wrapper = mount(RDialog, {
      propsData: {
        origin: 'top right'
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });

  it('should render component with custom width (max-width) and match snapshot', () => {
    const wrapper = mount(RDialog, {
      propsData: {
        maxWidth: 100
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });

  it('should render component with custom transition and match snapshot', () => {
    const wrapper = mount(RDialog, {
      propsData: {
        transition: 'fade-transition'
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });

  it('should open dialog on activator click', async () => {
    const input = jest.fn();
    const wrapper = mount(RDialog, {
      slots: {
        activator: [ compileToFunctions('<span>activator</span>') ]
      }
    });

    wrapper.vm.$on('input', input);

    expect(wrapper.vm.isActive).toBe(false);
    wrapper.find('.dialog__activator')[0].trigger('click');
    expect(wrapper.vm.isActive).toBe(true);
    await wrapper.vm.$nextTick();
    expect(input).toBeCalledWith(true);

    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });

  it('not should open disabed dialog on activator click', async () => {
    const input = jest.fn();
    const wrapper = mount(RDialog, {
      propsData: {
        disabled: true
      },
      slots: {
        activator: [ compileToFunctions('<span>activator</span>') ]
      }
    });

    wrapper.vm.$on('input', input);

    expect(wrapper.vm.isActive).toBe(false);
    wrapper.find('.dialog__activator')[0].trigger('click');
    expect(wrapper.vm.isActive).toBe(false);
    await wrapper.vm.$nextTick();
    expect(input).not.toBeCalled();

    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });

  it('not change state on v-model update', async () => {
    const wrapper = mount(RDialog, {
      propsData: {
        value: false
      },
      slots: {
        activator: [ compileToFunctions('<span>activator</span>') ]
      }
    });

    expect(wrapper.vm.isActive).toBe(false);

    wrapper.setProps({
      value: true
    });
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.isActive).toBe(true);

    wrapper.setProps({
      value: false
    });
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.isActive).toBe(false);

    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });

  it('should emit keydown event', async () => {
    const keydown = jest.fn();
    const component = {
      render: h => h(RDialog, {
        props: {
          value: true
        },
        on: {
          keydown
        }
      })
    };
    mount(component);

    window.dispatchEvent(new Event('keydown'));
    expect(keydown).toBeCalled();

    expect('Unable to locate target [data-app]').toHaveBeenTipped();
  });
});
