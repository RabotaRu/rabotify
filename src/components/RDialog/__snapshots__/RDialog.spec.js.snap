// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RDialog.vue should render a disabled component and match snapshot 1`] = `

<div class="dialog__container"
     style="display: block;"
>
  <div class="dialog__content"
       tabindex="-1"
       style="z-index: 0;"
  >
    <div class="dialog"
         style="display: none;"
    >
    </div>
  </div>
</div>

`;

exports[`RDialog.vue should render a fullscreen component and match snapshot 1`] = `

<div class="dialog__container"
     style="display: block;"
>
  <div class="dialog__content"
       tabindex="-1"
       style="z-index: 0;"
  >
    <div class="dialog dialog_fullscreen"
         style="display: none;"
    >
    </div>
  </div>
</div>

`;

exports[`RDialog.vue should render a lazy component and match snapshot 1`] = `

<div class="dialog__container"
     style="display: block;"
>
  <div class="dialog__content"
       tabindex="-1"
       style="z-index: 0;"
  >
    <div class="dialog"
         style="display: none;"
    >
    </div>
  </div>
</div>

`;

exports[`RDialog.vue should render a persistent component and match snapshot 1`] = `

<div class="dialog__container"
     style="display: block;"
>
  <div class="dialog__content"
       tabindex="-1"
       style="z-index: 0;"
  >
    <div class="dialog dialog_persistent"
         style="display: none;"
    >
    </div>
  </div>
</div>

`;

exports[`RDialog.vue should render a scrollable component and match snapshot 1`] = `

<div class="dialog__container"
     style="display: block;"
>
  <div class="dialog__content"
       tabindex="-1"
       style="z-index: 0;"
  >
    <div class="dialog dialog_scrollable"
         style="display: none;"
    >
    </div>
  </div>
</div>

`;

exports[`RDialog.vue should render component and match snapshot 1`] = `

<div class="dialog__container"
     style="display: block;"
>
  <div class="dialog__content"
       tabindex="-1"
       style="z-index: 0;"
  >
    <div class="dialog"
         style="display: none;"
    >
    </div>
  </div>
</div>

`;

exports[`RDialog.vue should render component with custom origin and match snapshot 1`] = `

<div class="dialog__container"
     style="display: block;"
>
  <div class="dialog__content"
       tabindex="-1"
       style="z-index: 0;"
  >
    <div class="dialog"
         style="display: none;"
    >
    </div>
  </div>
</div>

`;

exports[`RDialog.vue should render component with custom transition and match snapshot 1`] = `

<div class="dialog__container"
     style="display: block;"
>
  <div class="dialog__content"
       tabindex="-1"
       style="z-index: 0;"
  >
    <div class="dialog"
         style="display: none;"
    >
    </div>
  </div>
</div>

`;

exports[`RDialog.vue should render component with custom width (max-width) and match snapshot 1`] = `

<div class="dialog__container"
     style="display: block;"
>
  <div class="dialog__content"
       tabindex="-1"
       style="z-index: 0;"
  >
    <div class="dialog"
         style="max-width: 100px; display: none;"
    >
    </div>
  </div>
</div>

`;
